File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {

        vector<int>p(nums.size() , 1);
        vector<int>s(nums.size() , 1 );

        for(int i = 1 ; i < nums.size() ; i++)
        {
            p[i] = p[i-1] * nums[i-1];
        }
        for(int i = nums.size() - 2 ; i >= 0 ;i--)
        {
            s[i] = s[i+1] * nums[i+1];
        }

        vector<int>ans(nums.size() , 1);
        for(int i = 0 ; i < nums.size() ;i++)
        {
            ans[i] = p[i] * s[i];
        }

        return ans;
    }
};

File: Data Structures & Algorithms/sort-an-array/submission-0.java
class Solution {
    public int[] sortArray(int[] nums) {
        

         mS(nums , 0 , nums.length -1);

         return nums;
    }

    void merge(int[] nums , int low , int mid ,int high)
    {
        int left = low ;
        int right = mid + 1;

        ArrayList<Integer> temp = new ArrayList();

        while(left<= mid && right <= high)
        {
            if(nums[left] <= nums[right])
            {
                    temp.add(nums[left]);
                    left++;
            }
            else{
                temp.add(nums[right]);
                right++;
            }
        }


        while(left<=mid)
        {
            temp.add(nums[left]);
            left++;
        }

          while(right<=high)
        {
            temp.add(nums[right]);
            right++;
        }


           for (int i = low; i <= high; i++) {
            nums[i] = temp.get(i - low);
        }

    }
    void mS(int[] nums , int low , int high)
    {

        if(low >= high)return ;

        int mid = low + (high - low)/2;

        mS(nums , low , mid);
        mS(nums , mid + 1 , high);


        merge(nums , low , mid , high);

    }
}